modules/apps/object/object-test/src/testFunctional/macros/ValidateObjectsUpgrade.macro

```diff
@@ -0,0 +1,92 @@
+definition {
+
+	macro addEmailNotificationTemplate {
+		Navigator.openURL();
+
+		ApplicationsMenu.gotoPortlet(
+			category = "Notifications",
+			panel = "Control Panel",
+			portlet = "Templates");
+
+		ObjectNotifications.addNewNotificationTemplate(
+			bcc = "${bcc}",
+			cc = "${cc}",
+			description = "${description}",
+			emailBody = "${emailBody}",
+			entryTo = "${entryTo}",
+			fromAddress = "${fromAddress}",
+			fromName = "${fromName}",
+			name = "${name}",
+			notificationType = "${notificationType}",
+			subject = "${subject}");
+
+		if (isSet(objectLabelAttachment)) {
+			ObjectNotifications.addNotificationTemplateAttachment(objectLabel = "${objectLabelAttachment}");
+		}
+
+		PortletEntry.save();
+	}
+
+	macro editNotificationTemplate {
+		Navigator.openURL();
+
+		ApplicationsMenu.gotoPortlet(
+			category = "Notifications",
+			panel = "Control Panel",
+			portlet = "Templates");
```

Comment:
In general, we should call the navigation steps in a separate macro. The reasoning here is that there are some use cases where we are already on the page and we want to edit the template next. As is, we're going to add an unnecessary step to the test.